ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -663,15 +575,8 @@ def _handle_websocket_disconnect_message(
             call_trace=False,
             activate=True,
             tags={COMPONENT: self.integration_config.integration_name, SPAN_KIND: SpanKind.CONSUMER},
-        ) as ctx, ctx.span as disconnect_span:
-            disconnect_span.set_link(
-                trace_id=request_span.trace_id,
-                span_id=request_span.span_id,
-                attributes={SPAN_LINK_KIND: SpanLinkKind.EXECUTED},
-            )
-
-            if self.integration_config.asgi_websocket_messages_inherit_sampling:
-                _inherit_sampling_tags(disconnect_span, request_span._local_root)
+        ) as ctx:
+            core.dispatch("asgi.websocket.disconnect.message", (ctx, scope, message))
 
-            _copy_trace_level_tags(disconnect_span, request_span)
-            _set_websocket_close_tags(disconnect_span, message)
+        if request_span and request_span.error == 0:
+            request_span.finish()
```

Comment:
This seems out of place given that it's now nearly the only direct operation on a span in this file. Could change it in a separate pull request if you want.